lib/command_runner.dart

```diff
@@ -414,20 +420,43 @@ String _getCommandUsage(Map<String, Command> commands,
 
   // Show the commands alphabetically.
   names = names.toList()..sort();
+
+  // Group the commands by category.
+  var commandsByCategory = <String, List<Command>?>{};
```

Comment:
```suggestion
  var commandsByCategory = <String, List<Command>>{};
```

Reply:
Nit: You could instead use a [SplayTreeMap](https://api.dart.dev/stable/2.13.4/dart-collection/SplayTreeMap-class.html) here and then you wouldn't have to sort the keys later on (it would iterate in sorted order by key).

Reply:
Neat. Done